src/DotNetWorker.ApplicationInsights/Initializers/WebJobsRoleEnvironmentTelmetryInitializer.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.ApplicationInsights.Extensibility.Implementation;
+
+namespace Microsoft.Azure.Functions.Worker.ApplicationInsights.Initializers
+{
+    // This class was taken largely from https://raw.githubusercontent.com/Microsoft/ApplicationInsights-dotnet-server/91016d62f3181e10d4cf589ef8fd64dadb6b54a2/Src/WindowsServer/WindowsServer.Shared/AzureWebAppRoleEnvironmentTelemetryInitializer.cs, 
+    // but refactored so that it did not use WEBSITE_HOSTNAME, which is determined to be unreliable for functions during slot swaps.
+
+    /// <summary>
+    /// A telemetry initializer that will gather Azure Web App Role Environment context information.
+    /// </summary>    
+    internal class FunctionsRoleEnvironmentTelemetryInitializer : ITelemetryInitializer
+    {
+        internal const string AzureWebsiteName = "WEBSITE_SITE_NAME";
+        internal const string AzureWebsiteSlotName = "WEBSITE_SLOT_NAME";
+        internal const string AzureWebsiteCloudRoleName = "WEBSITE_CLOUD_ROLENAME";
+        private const string DefaultProductionSlotName = "production";
+        private const string WebAppSuffix = ".azurewebsites.net";
+
+        private readonly ConcurrentDictionary<string, string> _siteNodeNames = new(StringComparer.OrdinalIgnoreCase);
+
+        /// <summary>
+        /// Initializes <see cref="ITelemetry" /> device context.
+        /// </summary>
+        /// <param name="telemetry">The telemetry to initialize.</param>
+        public void Initialize(ITelemetry telemetry)
+        {
+            if (telemetry == null)
+            {
+                return;
+            }
+
+            var siteSlotName = new Lazy<string?>(() =>
```

Comment:
I see this is copied over. But it doesn't serve a purpose? What difference does reading this environment variable a few lines later matter?

``` CSharp
string siteSlotName = GetAzureWebsiteUniqueSlotName();
```
Is just as effective. It will be read independently for each `Initialize` call.

But given this is copied over, leave it if you want. Although it is an extra unnecessary allocation.

Reply:
I was wondering the same thing... I went back and it looks like @mathewc did this in a perf improvement PR several years ago: https://github.com/Azure/azure-webjobs-sdk/pull/2633. 

I know that Environment reads are abnormally expensive in Antares so he was likely only trying to read the env if it was absolutely necessary, even if it meant an extra allocation of that Lazy?

But yeah -- I mostly just copied it over as-is and didn't want to go mucking with it. Maybe we can refine it again, though...